mlibrary/cozy-sun-bear
# CSB-243/246: navigation slider stops updating

HOW TO TEST: EPUB

The navigator is more sensitive to EPUB locations: originally it treated the range as 1-100 and mapped percentages to locations, but this left the navigator extremely imprecise for keyboard navigation and syncing the range with the current location.

- open EPUB
- tab to focus on navigator
- arrow-right should advance the location
- arrow-left should backtrack the location
- tab out
- advancing by any other mean (next/prev buttons, etc) will continue updating the navigator

Note: because there are potentially many locations displayed on the viewport, arrow navigation can update the location status without updating the viewport.

HOW TO TEST: PDF

The navigator remains backwards-compatible with the PDFReader  location emulation layer.

- open PDF
- tab to focus on navigator
- arrow-right should advance the location
- arrow-left should backtrack the location
- tab out
- advancing by any other mean (next/prev buttons, etc) will continue updating the navigator
- scrolling the PDF area will update the navigator
